sse_starlette/sse.py

```diff
@@ -212,6 +220,13 @@ async def listen_for_exit_signal() -> None:
 
         # Await the event
         await AppStatus.should_exit_event.wait()
+        if using_uvicorn:
+            if graceful_shutdown_timeout is not None:
+                await anyio.sleep(graceful_shutdown_timeout)
+            else:
+                # Uvicorn waits indefinitely for tasks to complete if
+                # graceful_shutdown_timeout is None.
+                await anyio.sleep_forever()
```

Comment:
Here I am not sure: 
Even if this reflects default uvicorn behaviour, is this really what you would expect as a developer? I personally tend to rather stick to the existing behaviour: shut down immediately.